pkgs/by-name/ta/tabby-terminal/package.nix

```diff
@@ -0,0 +1,258 @@
+        desktopName = "Tabby";
+        exec = "tabby %u";
+        comment = "A terminal for a more modern age";
+        icon = "tabby";
+        categories = [
+          "Utility"
+          "TerminalEmulator"
+          "System"
+        ];
+      })
+    ];
+
+    electronPackage = electron_32;
+
+    electronHeaders = fetchurl {
+      url = "https://www.electronjs.org/headers/v${electronPackage.version}/node-v${electronPackage.version}-headers.tar.gz";
+      hash = "sha256-D9j1wSDzartWUfDm2UrZDrgum2X+vV+DpDlKPFCtBbA=";
+    };
+
+    electronHeadersSHA = fetchurl {
+      url = "https://www.electronjs.org/headers/v${electronPackage.version}/SHASUMS256.txt";
+      hash = "sha256-ZnWVCzTwwog4kgeFAZGsM2F2mmebrpXMVMxn/K+LWlU=";
+    };
+
+    buildInputs = [
+      nodejs
+      python3
+      fontconfig
+      electronPackage
+    ];
+
+    nativeBuildInputs = [
+      git
+      yarn
+      patch-package
+      prefetch-yarn-deps
+      fixup-yarn-lock
+      (python3.withPackages (python-pkgs: [
+        python-pkgs.distutils
+        python-pkgs.gyp
+      ]))
+      makeWrapper
+      http-server
+      pkg-config
+      libsecret
+      copyDesktopItems
+      jq
+      moreutils
+    ];
+
+    patchPhase = ''
+      runHook prePatch
+
+      jq '.version = "${version}"' app/package.json | sponge app/package.json
+
+      runHook postPatch
+    '';
+
+    configurePhase =
+      ''
+        runHook preConfigure
+
+        export buildDir=$PWD
+
+        git tag ${version}
+      ''
+      # Loop through all the yarn caches and install the deps for the respective package
+      + builtins.concatStringsSep "\n" (
+        map (cache: ''
+          cd ${cache.pkg}
+          export HOME=$PWD/yarn_home
+          fixup-yarn-lock yarn.lock
+          yarn config --offline set yarn-offline-mirror ${cache.cache}
+          echo "Installing deps for ${cache.pkg}"
+          yarn install --offline --prefer-offline --frozen-lockfile --ignore-scripts --no-progress --non-interactive
+          patch-package
+          patchShebangs node_modules
+          echo "Done!"
+          cd $buildDir
+        '') yarnCaches
+      )
+      + ''
+        cd $buildDir/node_modules
+      ''
+      # Loop thought the "built in" plugins and link them to the node_modules
+      + builtins.concatStringsSep "\n" (
+        map (plugin: ''
+          ln -fs ../${plugin} ${plugin}
+        '') builtinPlugins
+      )
+      + ''
+        cd $buildDir
+
+        runHook postConfigure
+      '';
+
+    buildPhase =
+      # Start a fake (and completely unnecessary) http-server to let electron-rebuild "download" the headers and hashes
+      ''
+        runHook preBuild
+
+        mkdir -p http-cache/v${electronPackage.version}
+        cp $electronHeaders http-cache/v${electronPackage.version}/node-v${electronPackage.version}-headers.tar.gz
+        cp $electronHeadersSHA http-cache/v${electronPackage.version}/SHASUMS256.txt
+        http-server http-cache &
```

Comment:
I think `__darwinAllowLocalNetworking = true;` needs to be set for it to compile on Darwin (I know you're not adding Darwin support yet but it's still good to set this IMO as a reminder)